base/markdown/Common/block.jl

```diff
@@ -198,7 +198,7 @@ function list(stream::IO, block::MD)
                 c = read(stream, Char)
                 if c == '\n'
                     eof(stream) && break
-                    next = peek(stream)
+                    next = Char(peek(stream)) # ok since we only compare with ASCII
                     if next == '\n'
```

Comment:
Performance-wise, both solutions seem to give the same native code. I just tried with:
```julia
function f(x)
     for c in x
         Char(c) == '\n' && return true
     end
     false
 end

function g(x)
     for c in x
         c == UInt8('\n') && return true
     end
     false
 end

code_native(f, (Vector{UInt8},))
code_native(g, (Vector{UInt8},))
```

But as regards style, I prefer converting to `Char`.